spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -525,6 +525,7 @@ protected String errorsBaseName(ProducerDestination destination) {
 		return destination.getName() + ".errors";
 	}
 
+	@Deprecated
 	private final class ReceivingHandler extends AbstractReplyProducingMessageHandler {
```

Comment:
I wonder if we can remove this internal classes altogether. 
We are in major version, so any braking changes like this is possible.
Or do you leave them for the legacy data support?

Reply:
I think we may be able to remove it, haven't really thought much about it. I know for a fact that once we take care of Embedding headers as an interceptor too, there won't be much left in those classes.
I just tagged them to keep an eye out during the epic of 2.0 redesign